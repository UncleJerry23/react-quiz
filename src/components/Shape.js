import React from 'react';
import PropTypes from 'prop-types';

export default function Shape({ text, textColor, blockColor }) {

  const textStyle = {
    color: { textColor }
  };

  const divStyle = {
    backgroundImage: { blockColor },
    width: '100px',
    height: '100px'
  };

  return (
    <>
      <p style={textStyle}>{text}</p>,
      <div style={divStyle}></div>
    </>
  );
}

Shape.propTypes = {
  text: PropTypes.string.isRequired,
  textColor: PropTypes.string.isRequired,
  blockColor: PropTypes.string.isRequired
};

