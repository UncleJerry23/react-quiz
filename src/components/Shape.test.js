import React from 'react';
import { shallow } from 'enzyme';
import Shape from './Shape';

describe('Shape component', () => {
  it('renders a Shape', () => {
    const wrapper = shallow(<Shape text='hi' textColor='#000000' blockColor='#000000'/>);
    expect(wrapper).toMatchSnapshot();
  });
});
