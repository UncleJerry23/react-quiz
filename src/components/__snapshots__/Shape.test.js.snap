// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape component renders a Shape 1`] = `
<Fragment>
  <p
    style={
      Object {
        "color": Object {
          "textColor": "#000000",
        },
      }
    }
  >
    hi
  </p>
  ,
  <div
    style={
      Object {
        "backgroundImage": Object {
          "blockColor": "#000000",
        },
        "height": "100px",
        "width": "100px",
      }
    }
  />
</Fragment>
`;
